text_extract.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+from unittest import mock
+from unittest.mock import patch, MagicMock
+import pandas as pd
+from extract import get_plant_data, extract_plant_data, load_into_dataframe, TOTAL_NUMBER_OF_PLANTS
+
+
+@pytest.fixture
+def initial_dataframe():
+    return pd.DataFrame({
+        "botanist_first_name": pd.Series(dtype=str),
+        "botanist_last_name": pd.Series(dtype=str),
+        "botanist_email": pd.Series(dtype=str),
+        "botanist_phone_number": pd.Series(dtype=str),
+        "plant_name": pd.Series(dtype=str),
+        "plant_scientific_name": pd.Series(dtype=str),
+        "recording_taken": pd.Series(dtype=str),
+        "last_watered": pd.Series(dtype=str),
+        "soil_moisture": pd.Series(dtype=float),
+        "temperature": pd.Series(dtype=float),
+        "country_code": pd.Series(dtype=str),
+    })
+
+
+def mock_get_success(url):
+    """Mock a successful API response"""
+    return mock.Mock(status_code=200, json=lambda: {"name": "Epipremnum Aureum"})
+
+
+def test_get_plant_data_success():
+    with mock.patch("requests.get", side_effect=mock_get_success):
```

Comment:
might be overkill since the function names are pretty descriptive already, but do we want docstrings for all tests?